x-pack/plugins/infra/server/lib/alerting/metric_threshold/register_metric_threshold_alert_type.ts

```diff
@@ -100,7 +85,114 @@ export function registerMetricThresholdAlertType(libs: InfraBackendLibs): Metric
     defaultActionGroupId: FIRED_ACTIONS.id,
     actionGroups: [FIRED_ACTIONS, WARNING_ACTIONS],
     minimumLicenseRequired: 'basic',
-    executor: createMetricThresholdExecutor(libs),
+    executor: async ({ services, params }) => {
```

Comment:
As I indicated in a recent conversation, I'm not a fan of dumping all of this in here. It breaks the unit test and makes it a lot harder to test in isolation at all. It also creases the risk that we're closing over some variable unintentionally, which could lead to hard-to-diagnose bugs. If you don't mind I'll try to tease them apart again.

Reply:
My intention was absolutely to get them apart again, I just needed to see if it worked at all. So yes, please tease away :)